USM.Test/AlgorithmTest.cs

```diff
@@ -0,0 +1,509 @@
+        public void TestNoConflictDifDayDifTime()
+        {
+            //use the course above
+            ScheduleRepo sch = new ScheduleRepo();
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE20 = AllCourses[0];
+            Course CSE25 = AllCourses[5];
+            Course[] courses = { CSE20, CSE25 };
+
+            //call the algorithm
+            List<List<Section>> returned = sch.FindScheduleForClasses(courses);
+            int expectedNum = 1;
+            int actualNum = returned.Count;
+
+            //two schedules
+            Assert.Equal(expectedNum, actualNum);
+        }
+
+        // no conflict with two discussions in one class
+        [Fact]
+        public void TestNoConflictWithDis1()
+        {
+            //use the course above
+            ScheduleRepo sch = new ScheduleRepo();
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE20 = AllCourses.First(c => c.CourseAbbreviation == "CSE20");
+            Course CSE26 = AllCourses.First(c => c.CourseAbbreviation == "CSE26");
+            Course[] courses = { CSE20, CSE26 };
+
+            //call the algorithm
+            List<List<Section>> returned = sch.FindScheduleForClasses(courses);
+            int expectedNum = 2;
+            int actualNum = returned.Count;
+
+            //two schedules
+            List<Section> ls0 = returned[0];
+            Section dis1 = ls0[1];
+            List<Section> ls1 = returned[1];
+            Section dis2 = ls1[1];
+            Debug.WriteLine("first dis is "+dis1.Course.CourseAbbreviation);
+            Debug.WriteLine("second dis is " + dis2.Course.CourseAbbreviation);
+
+
+            
+            Assert.Equal(expectedNum, actualNum);
+        }
+
+        //one discussion pass but one discussion conflicts
+        [Fact]
+        public void TestNoConflictWithDis2()
+        {
+            //use the course above
+            ScheduleRepo sch = new ScheduleRepo();
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE24 = AllCourses[4];
+            Course CSE26 = AllCourses[6];
+            Course[] courses = { CSE24, CSE26 };
+
+            //call the algorithm
+            List<List<Section>> returned = sch.FindScheduleForClasses(courses);
+            int expectedNum = 1;
+            int actualNum = returned.Count;
+
+            //two schedules
+            Assert.Equal(expectedNum, actualNum);
+            //test whether the right list 
+            List<List<Section>> possibleSchedule = returned;
+            List<Section> rightSchedule = new List<Section>();
+            rightSchedule.Add(CSE24.Sections.ElementAt(0));
+            rightSchedule.Add(CSE26.Sections.ElementAt(1));
+            Assert.Contains(rightSchedule, returned);
+        }
+
+        // one confliction between discussion and discussion
```

Comment:
This is a good example of when you should name the courses to represent what conflicts occur from them.  Which classes conflict?